### ISSUE TITLE
Add support for `clear` method on final TypedDict when all keys are `NotRequired`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#4794]
See [this issue](https://github.com/python/mypy/issues/12732) and [this issue](https://github.com/python/mypy/issues/14914#issuecomment-1473559078) in the mypy issue tracker.

**Comment by erictraut** [comment]
This will be included int he next release.

**Comment by erictraut** [comment]
I added `popitem(self) -> tuple[str, Unknown]`. This is more aligned with existing TypedDict functionality and probably more useful than generating a union of all value types.

**Comment by alicederyn** [comment]
won't that mean you can't do this?
```python
if v[0] == "key1":
  ... // code that now knows the type of v[1]
```

**Comment by erictraut** [comment]
Yes, there are tradeoffs to the various choices. As I mentioned, `tuple[str, Unknown]` is more consistent with other choices we've made in pyright when it comes to TypedDict.

**Comment by erictraut** [comment]
This is included in pyright 1.1.300, which I just published. It will also be included in a future release of pylance.
